src/utils.rs

```diff
@@ -50,3 +58,35 @@ pub fn parse_gj_messages_response(meow: String) -> Result<Vec<GDMessage>, Generi
         })
         .collect()
 }
+
+pub fn valid_url_check(url: &str) -> bool {
+    let reg = regex::Regex::new(r#"^(http://www\.|https://www\.|http://|https://)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?$"#).unwrap();
+
+    if reg.is_match(&url.to_lowercase()) {
+        true
+    } else if reg.is_match(&url.to_lowercase()) {
+        true
+    } else {
+        println!("\x1b[0;31mInvalid URL ({})\x1b[0m", url);
+        false
+    }
+}
+
+pub async fn proxy_list() -> Result<String, GenericError> {
+    let srv: String = if std::fs::File::open("servers.json").is_err() {
+        BOOMLINGS_SERVER.to_string()
+    } else {
+        let file = std::fs::read_to_string("servers.json")?;
+        let s: Servers = serde_json::from_str(&file).unwrap();
+        let count = s.servers.len();
+        let num = rand::thread_rng().gen_range(0..count);
+
+        if !s.servers[num].is_empty() && valid_url_check(&*s.servers[num]) {
+        }
```

Comment:
why is there a pointless if statement here

Reply:
this if statement is not pointless. If the servers.json is not empty, but the url check fails, it will cry (see attached picture). If both checks are successful it will just continue without crying.
![image](https://github.com/RoootTheFox/dashend/assets/59457929/d60ee2d4-3e98-49e3-939c-20adf2733b48)

Reply:
But i am changing it up anyways, so don't resolve just yet